ci/Dockerfile

```diff
@@ -0,0 +1,13 @@
+# Build the correct binaries
+FROM fossa/haskell-static-alpine:ghc-8.10.4 as builder
+RUN cabal update
+RUN cabal install hlint
+RUN cabal install fourmolu
```

Comment:
Not pinning these spooks me a little. This build is not reproducible. Can we pin the `hlint` and `fourmolu` versions, at least?

Reply:
We just pull the built image, we don't rebuild this every time.  It's reproducible for that reason.  I think pinning here is also not the right move, I'd prefer to always use the latest, I just didn't want to pay the cost of installing these every time.

In this specific case, what does reproducibility gain us?

Reply:
> we don't rerun this dockerfile every time

This is _exactly_ the problem. Someone who runs these commands locally may end up with a different set of tool versions than the tools being used in CI.

Pinning here makes it _extremely_ obvious which tool versions are being used.

I think we should pin because:

1. Pinning provides value - the tool versions become obvious.
2. Not pinning causes subtle problems - the tool versions can drift silently.
3. Not pinning does not provide significantly more value than pinning - you can simulate "always use latest" by just moving the pin.
4. There's really no cost here to pinning. The only cost I can think of is "we're not always on latest", but as you've mentioned, we're _already_ not on latest even without pinning. The only difference is that now we're on a _known_ version of not-latest instead of an _arbitrary depending on when the Docker build runs_ version of not-latest.

Being explicit is better than being magic.